### PR TITLE
Git hooks, github templates, and code climate

### PR DESCRIPTION
**Pull request by argos83**

### CODE REVIEW

File: tests/__init__.py [hunk 1]
```diff
@@ -19,3 +22,15 @@ def env_var(name, value):
             os.environ.pop(name)
         else:
             os.environ[name] = old_value
+
+
+class CaptureOutput(list):
```
Comment (estemendoza) [inline tests/__init__.py]: Can you explain what is this for?

Comment (estemendoza) [inline tests/__init__.py]: Sorry, I just realized it what is this for

**Comment by argos83** [comment]
> Can you explain the github hooks?

If you meant the github templates: here they are explained: https://github.com/blog/2111-issue-and-pull-request-templates

If you meant the [git hooks](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks), those are scripts which you need to place in your .git/hooks/ directory (I added this `bin/setup-env` script that does it for you). One of them will check your commit messages and fail if they message format is not compliant with the conventional commits), the other will run the unit tests and abort the commit if they fail, and the other one will run the tests in all the supported platforms when you try to push, and abort the push if they fail.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+Marvin uses [flake8][] to lint both Marvin and tests codebases. Execute `flake8 marvin tests`.
+
+### Documentation
+
+The main documentation is written in [Markdown][]. Marvin uses
+[ReadTheDocs][] to build and publish the documentation:
+
+- [https://marvin.readthedocs.io](https://marvin.readthedocs.io) - preferred long URL
+- [http://marvin.rtfd.org](http://marvin.rtfd.org) - preferred short URL
```
Comment (estemendoza) [inline .github/CONTRIBUTING.md]: I think these two url links to the wrong python project, do we have a valid url for readthedocs?

Comment (argos83) [inline .github/CONTRIBUTING.md]: I don't know, I just wrote those as placeholders. We should register an account there.

Comment (argos83) [inline .github/CONTRIBUTING.md]: Yes, marvin is taken. I've just registered marvin-test. I'll update this PR

**Pull request merged**